src/parsers/manifest/dash/parsers_types.ts

```diff
@@ -80,6 +80,8 @@ export interface ILoadedResource<T extends string | ArrayBuffer> {
    * `undefined` if unknown or not applicable.
    */
   receivedTime? : number;
-  /** The loaded resource itself, under the right format. */
-  responseData : T;
+  /** The loaded resource itself, under the right format.
+   * Or an error, when fetching ressources.
+   */
+  responseData : T | { onError: boolean; message: string };
```

Comment:
It's kind-of risky to have `unknown generic | defined object`, no?
And it's harder to check.

Why not a complete and idiomatic [discriminated union](https://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes-func.html#discriminated-unions)

Like:
```ts
response: {
  success: true;
  data: T;
} | {
  success: false;
  error: Error;
}
```